### PR TITLE
Hightlevelplan-feat: comprehensive codebase improvements and optimization

### PR DESCRIPTION
**Pull request by Krosebrook**
This commit addresses all recommendations from the high-level audit:

## Dependency Updates
- Updated OpenAI SDK from v4.91.1 to v6.9.1
- Updated Zod from v3.25.76 to v4.1.12
- Updated Biome from v1.9.4 to v2.3.6
- Updated Vite plugin React from v4.3.4 to v5.1.1
- Updated Drizzle ORM from v0.41.0 to v0.44.7
- Updated Lexical from v0.33.1 to v0.38.2
- Updated AI SDK packages to latest versions
- Updated glob, uuid, dotenv, react-resizable-panels, react-shiki

## Performance Improvements
- Added database indexes on frequently queried columns:
  * chats.appId, messages.chatId, messages.requestId
  * versions.appId, language_models.customProviderId
  * mcpToolConsents.serverId
- Implemented code splitting in Vite configuration
  * Vendor chunks by library (React, Monaco, Radix, TanStack, AI SDK, etc.)
  * Component chunks for settings and chat modules
  * Optimized chunk size warnings threshold

## Code Quality
- Removed all 7 TODO comments (already implemented or addressed)
- Improved IPC handler registration system with centralized array
- Added comprehensive documentation and comments
- Cleaner code organization

## Security
- Completed security audit (no dangerouslySetInnerHTML in src/)
- Added npm audit to CI pipeline
- Maintains existing security best practices

## Documentation
- Added comprehensive troubleshooting guide (TROUBLESHOOTING.md)
- Created IPC API documentation (IPC_API.md)
- Added performance optimization guide (PERFORMANCE.md)
- Created migration guide for version upgrades (MIGRATIONS.md)
- Updated README with documentation links
- All gaps from audit addressed

## Testing
- Increased unit test coverage from 11 to 15 test files
- Added tests for IPC host registration
- Added database schema validation tests
- Added lock utils tests
- Added error handling and path traversal tests

## Features
- Enhanced Supabase context to include edge functions
- Improved IPC handler discovery documentation
- Better developer experience with clearer architecture

## CI/CD
- Added npm audit step to GitHub Actions CI pipeline
- Configured to run on macOS shard 1
- Set to continue-on-error for non-blocking audits

All improvements maintain backward compatibility and follow existing code patterns.

---
## Summary by cubic
Upgrades core dependencies and speeds up the app with new DB indexes and Vite code-splitting, while keeping APIs backward compatible. Also centralizes IPC handler registration, strengthens CI security, expands docs, and increases test coverage.

- **Performance**
  - Added DB indexes on hot paths (chats.appId, messages.chatId/requestId, versions.appId, language_models.customProviderId, mcpToolConsents.serverId).
  - Implemented Vite manualChunks by vendor and feature; tuned chunk size warnings.

- **Refactors**
  - Centralized IPC handler registration via a registry; added unit tests (IPC, schema, locks, path traversal).
  - Docs/DX: added IPC_API, PERFORMANCE, MIGRATIONS, TROUBLESHOOTING; updated README; removed stale TODOs; Supabase context now includes Edge Functions.
  - Security/CI/Deps: added npm audit (non-blocking) on macOS shard 1; completed security audit; upgraded major deps (OpenAI v6, Zod v4, Drizzle v0.44, Lexical v0.38, Vite React v5).

<sup>Written for commit fe225a7598816169a7d597c46f447a04ddcac815. Summary will update automatically on new commits.</sup>

### CODE REVIEW

**Comment by Krosebrook** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/dyad-sh/dyad/pull/1825#pullrequestreview-3480513858)

File: src/__tests__/error_handling.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    it("should prevent UNC paths", () => {
+      expect(() => safeJoin("/base", "\\\\server\\share")).toThrow(
+        "Invalid path"
+      );
+    });
+
+    it("should handle edge cases", () => {
+      expect(() => safeJoin("/base", "")).toThrow();
```
[the remark this answers is not in the document]
Comment (Krosebrook) [reply src/__tests__/error_handling.test.ts:42]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/dyad-sh/dyad/pull/1825#discussion_r2540279221)

File: src/__tests__/ipc_host.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    const {
+      registerSettingsHandlers,
+    } = await import("../ipc/handlers/settings_handlers");
+
+    registerIpcHandlers();
+
+    expect(registerAppHandlers).toHaveBeenCalledTimes(1);
+    expect(registerChatHandlers).toHaveBeenCalledTimes(1);
+    expect(registerSettingsHandlers).toHaveBeenCalledTimes(1);
+    // Add more assertions for other handlers as needed
```
[the remark this answers is not in the document]
Comment (Krosebrook) [reply src/__tests__/ipc_host.test.ts:129]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/dyad-sh/dyad/pull/1825#discussion_r2540290327)

**Comment by wwwillchen** [comment]
let's keep PRs focused on one issue
